hivemind/moe/server/layers/optim.py

```diff
@@ -0,0 +1,63 @@
+import torch
+
+
+class OptimizerWrapper(torch.optim.Optimizer):
+    """A wrapper for pytorch.optim.Optimizer that forwards all methods to the wrapped optimizer"""
+
+    def __init__(self, optim: torch.optim.Optimizer):
+        object.__init__(self)
```

Comment:
object? In that case, it’s not a true Optimizer

Reply:
on the contrary it defines all optimizer fields and methods as properties